src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -496,182 +1249,315 @@ bool emberAfThermostatClusterSetActivePresetRequestCallback(
+        {
+            pendingPreset.SetCoolingSetpoint(MakeOptional(EnforceCoolingSetpointLimits(coolingSetpointValue.Value(), endpoint)));
+        }
 
-    // Note that the limits are initialized above per the spec limits
-    // if they are not present Get() will not update the value so the defaults are used
-    imcode status;
+        Optional<int16_t> heatingSetpointValue = preset.GetHeatingSetpoint();
+        if (heatingSetpointValue.HasValue())
+        {
+            pendingPreset.SetHeatingSetpoint(MakeOptional(EnforceHeatingSetpointLimits(heatingSetpointValue.Value(), endpoint)));
+        }
+    }
 
-    // https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/3724
-    // behavior is not specified when Abs * values are not present and user values are present
-    // implemented behavior accepts the user values without regard to default Abs values.
+    uint8_t totalCount = CountUpdatedPresetsAfterApplyingPendingPresets(delegate);
 
-    // Per global matter data model policy
-    // if a attribute is not present then it's default shall be used.
+    uint8_t numberOfPresetsSupported = delegate->GetNumberOfPresets();
 
-    status = AbsMinCoolSetpointLimit::Get(endpoint, &AbsMinCoolSetpointLimit);
-    if (status != imcode::Success)
+    if (numberOfPresetsSupported == 0)
     {
-        ChipLogError(Zcl, "Warning: AbsMinCoolSetpointLimit missing using default");
+        ChipLogError(Zcl, "emberAfThermostatClusterCommitPresetsSchedulesRequestCallback: Failed to get NumberOfPresets");
+        return SendResponseAndCleanUp(delegate, endpoint, commandObj, commandPath, imcode::InvalidInState);
     }
 
-    status = AbsMaxCoolSetpointLimit::Get(endpoint, &AbsMaxCoolSetpointLimit);
-    if (status != imcode::Success)
+    // If the expected length of the presets attribute with the applied changes exceeds the total number of presets supported,
+    // return RESOURCE_EXHAUSTED. Note that the changes are not yet applied.
+    if (numberOfPresetsSupported > 0 && totalCount > numberOfPresetsSupported)
     {
-        ChipLogError(Zcl, "Warning: AbsMaxCoolSetpointLimit missing using default");
+        return SendResponseAndCleanUp(delegate, endpoint, commandObj, commandPath, imcode::ResourceExhausted);
     }
 
-    status = MinCoolSetpointLimit::Get(endpoint, &MinCoolSetpointLimit);
-    if (status != imcode::Success)
-    {
-        MinCoolSetpointLimit = AbsMinCoolSetpointLimit;
-    }
+    // TODO: Check if the number of presets for each presetScenario exceeds the max number of presets supported for that
+    // scenario. We plan to support only one preset for each presetScenario for our use cases so defer this for re-evaluation.
 
-    status = MaxCoolSetpointLimit::Get(endpoint, &MaxCoolSetpointLimit);
-    if (status != imcode::Success)
+    // Call the delegate API to apply the pending presets to the presets attribute and update it.
+    err = delegate->ApplyPendingPresets();
```

Comment:
So here's the thing.  We set the preset handle to a "unique value" above, on a temporary object that has since gone out of scope.  What was the point of that?

Seems like what should actually happen is that ApplyPendingPresets is responsible for allocating unique handles, and the API should document that.  And the implementation should do it.

Reply:
yup i realized that as well. I will remove the unique value above and add it in applypendingpresets and document it

Reply:
done.